### PR TITLE
Modify resources and exception throws that contain paths

### PR DESCRIPTION
**Pull request by dj-wyder**

### CODE REVIEW

File: lib/TransferControllers/SyncTransferController.cs [hunk 1]
```diff
@@ -178,9 +180,10 @@ private async Task ValidateDestinationPathAsync()
 		        }
 		        catch (Exception ex)
 		        {
-			        throw new TransferInvalidPathException(string.Format(CultureInfo.CurrentCulture, 
-				        Resources.DestinationPathValidationFailed, transferLocation.ToString()), ex);
-		        }
+			        var exception = new TransferInvalidPathException(Resources.DestinationPathValidationFailed, ex);
+                    exception.Data.Add("path", transferLocation.ToString());
```
Comment (KlaudiuszBryjaRelativity) [inline lib/TransferControllers/SyncTransferController.cs:184]: Please fix the formatting

File: version.txt [hunk 2]
```diff
@@ -1 +1 @@
-4.3.1
\ No newline at end of file
+4.3.2
```
Comment (KlaudiuszBryjaRelativity) [inline version.txt:1]: Did you execute `tools\scripts\InjectBuildNumber.ps1` to apply this version? It looks like you did but I'd like to be sure.

**Review by KlaudiuszBryjaRelativity — APPROVED** [review]

File: version.txt (hunk 2 above)
Comment (dj-wyder) [reply version.txt:1]: I didn't, I made it manually by looking at other PRs. Does it do something more than modifying the files that contain version number?

Comment (KlaudiuszBryjaRelativity) [reply version.txt:1]: It does the same but for you it is only one place to run this script.

File: samples/Microsoft.Azure.Storage.DataMovement.Client/Transfers/UploadDirectoryTransfer.cs [hunk 3]
```diff
@@ -37,7 +37,15 @@ protected override void ValidateImpl()
             ValidateUrl(Options.Destination, nameof(Options.Destination), "Destination should be http(s) url.");
 
             if (!Directory.Exists(Options.Source))
-                throw new DirectoryNotFoundException($"Directory ({Options.Source}) does not exist.");
+            {
+                var ex = new TransferException( 
+                    TransferErrorCode.DirectoryNotFound, 
+                    "Directory ({Options.Source}) does not exist.");
```
Comment (KamilKalitaRel) [inline samples/Microsoft.Azure.Storage.DataMovement.Client/Transfers/UploadDirectoryTransfer.cs:43]: Shouldn't the path `Options.Source` be removed from the message?

**Review by KlaudiuszBryjaRelativity — APPROVED** [review]

**Review by KamilKalitaRel — APPROVED** [review]

**Pull request merged**